sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/AppConfigurationBootstrapConfiguration.java

```diff
@@ -152,21 +131,15 @@ public ClientStore buildClientStores(AppConfigurationProperties properties,
             clientProvider = clientProviderOptional.get();
         }
 
-        boolean isDev = false;
-        boolean isKeyVaultConfigured = false;
+        boolean isKeyVaultConfigured = keyVaultCredentialProviderOptional.isPresent() || keyVaultClientProviderOptional.isPresent();
 
-        for (String profile : env.getActiveProfiles()) {
-            if ("dev".equalsIgnoreCase(profile)) {
-                isDev = true;
-                break;
-            }
-        }
+        String clientId = "";
 
-        if (keyVaultCredentialProviderOptional.isPresent() || keyVaultClientProviderOptional.isPresent()) {
-            isKeyVaultConfigured = true;
+        if (properties.getManagedIdentity() != null) {
+            clientId = properties.getManagedIdentity().getClientId();
         }
 
-        return new ClientStore(appProperties, pool, tokenCredentialProvider, clientProvider, isDev,
-            isKeyVaultConfigured);
+        return new AppConfigurationReplicaClientsBuilder(tokenCredentialProvider, clientProvider, isKeyVaultConfigured,
+            clientId, appProperties.getMaxRetries());
```

Comment:
How about using a builder pattern here?
```java
new AppConfigurationReplicaClientsBuilder()
.tokenCredentialProvider(xxx)
.clientProvider(xxx)
.clientId(xxx)
.maxRetries(xxx)
```

Reply:
I am mostly ok with this change, but I think MaxRetries needs to be part of the constructor as it is always required.